solidity/contracts/hooks/PolygonPosHook.sol

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.8.0;
+
+/*@@@@@@@       @@@@@@@@@
+ @@@@@@@@@       @@@@@@@@@
+  @@@@@@@@@       @@@@@@@@@
+   @@@@@@@@@       @@@@@@@@@
+    @@@@@@@@@@@@@@@@@@@@@@@@@
+     @@@@@  HYPERLANE  @@@@@@@
+    @@@@@@@@@@@@@@@@@@@@@@@@@
+   @@@@@@@@@       @@@@@@@@@
+  @@@@@@@@@       @@@@@@@@@
+ @@@@@@@@@       @@@@@@@@@
+@@@@@@@@@       @@@@@@@@*/
+
+// ============ Internal Imports ============
+import {AbstractMessageIdAuthHook} from "./libs/AbstractMessageIdAuthHook.sol";
+import {StandardHookMetadata} from "./libs/StandardHookMetadata.sol";
+import {TypeCasts} from "../libs/TypeCasts.sol";
+import {Message} from "../libs/Message.sol";
+import {IPostDispatchHook} from "../interfaces/hooks/IPostDispatchHook.sol";
+
+// ============ External Imports ============
+import {FxBaseRootTunnel} from "fx-portal/tunnel/FxBaseRootTunnel.sol";
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+
+/**
+ * @title PolygonPosHook
+ * @notice Message hook to inform the PolygonPosIsm of messages published through
+ * the native PoS bridge.
+ */
+contract PolygonPosHook is AbstractMessageIdAuthHook, FxBaseRootTunnel {
+    using StandardHookMetadata for bytes;
+
+    // ============ Constructor ============
+
+    constructor(
+        address _mailbox,
+        uint32 _destinationDomain,
+        bytes32 _ism,
+        address _cpManager,
+        address _fxRoot
+    )
+        AbstractMessageIdAuthHook(_mailbox, _destinationDomain, _ism)
+        FxBaseRootTunnel(_cpManager, _fxRoot)
+    {
+        require(
+            Address.isContract(_cpManager),
+            "PolygonPosHook: invalid cpManager contract"
+        );
+        require(
+            Address.isContract(_fxRoot),
+            "PolygonPosHook: invalid fxRoot contract"
+        );
```

Comment:
does `FxBaseRootTunnel` not do this?

Reply:
No it's not checked in `FxBaseRootTunnel`. It construct with below 2 interfaces
```
constructor(address _checkpointManager, address _fxRoot) {
        checkpointManager = ICheckpointManager(_checkpointManager);
        fxRoot = IFxStateSender(_fxRoot);
    }
```